cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js

```diff
@@ -0,0 +1,444 @@
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import WrangleHistory from 'components/Wrangler/WrangleHistory';
+import {inferColumn} from 'components/Wrangler/type-inference';
+import classnames from 'classnames';
+import shortid from 'shortid';
+import {
+  dropColumn,
+  renameColumn,
+  splitColumn,
+  mergeColumn
+} from 'components/Wrangler/column-transforms';
+
+export default class WrangleData extends Component {
+  constructor(props) {
+    super(props);
+
+    let headers = Object.keys(this.props.data[0]);
+
+    let columnTypes = {};
+    headers.forEach((column) => {
+      columnTypes[column] = inferColumn(this.props.data, column);
+    });
+
+    this.state = {
+      headersList: headers,
+      data: this.props.data,
+      history: [],
+      columnTypes: columnTypes,
+      activeSelection: null,
+      activeSelectionType: null,
+      isRename: false,
+      isSplit: false,
+      isMerge: false
+    };
+  }
+
+  renderActionList() {
+    if (this.state.activeSelectionType === 'COLUMN') {
+      return this.renderColumnActions();
+    }
+  }
+
+  renderColumnActions() {
+    return (
+      <div className="btn-group-vertical">
+        <button
+          className="btn btn-default"
+          onClick={this.dropColumn.bind(this)}
+        >
+          Drop column
+        </button>
+        <button
+          className="btn btn-default"
+          onClick={this.splitColumnClick.bind(this)}
+        >
+          Split column
+        </button>
+        {this.renderSplit()}
+
+        <button
+          className="btn btn-default"
+          onClick={this.mergeColumnClick.bind(this)}
+        >
+          Merge column
+        </button>
+        {this.renderMerge()}
+
+        <button
+          className="btn btn-default"
+          onClick={this.renameColumnClick.bind(this)}
+        >
+          Rename column
+        </button>
+        {this.renderRename()}
+      </div>
+    );
+  }
+
+  columnClickHandler(column) {
+    this.setState({
+      activeSelectionType: 'COLUMN',
+      activeSelection: column
+    });
+  }
+
+  renderRename() {
+    if (!this.state.isRename) { return null; }
+
+    return (
+      <div className="rename-input">
+        <label className="label-control">New name</label>
+        <input
+          type="text"
+          className="form-control"
+          ref={(ref) => this.renameInput = ref}
```

Comment:
Why not use `onChange`? We should not use refs to get value of a textbox. We have an onChange handler that updates the state